### PR TITLE
Add analysis types and names list CORRECTED TESTS

### PR DESCRIPTION
**Pull request by KapRom**
corrected PR #176

### CODE REVIEW

File: src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {MatListModule, MatExpansionModule, MatButtonModule, MatCardModule} from '@angular/material';
+import { AnalysisTypesListComponent } from "../analysis-types-list/analysis-types-list.component";
+import {} from 'jasmine';
```
Comment (Sebaxp95) [inline src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts]: Empty import

File: src/app/analysis-view/analysis-view/analysis-view.component.spec.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ import 'rxjs/add/observable/of';
 
 import {
   MatButtonModule,
-  MatCardModule,
+  MatCardModule, MatListModule,
```
Comment (Sebaxp95) [inline src/app/analysis-view/analysis-view/analysis-view.component.spec.ts]: With Grzegorz style it should be in new line

File: src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+  selector: 'app-analysis-names-list',
+  templateUrl: './analysis-names-list.component.html',
+  styleUrls: ['./analysis-names-list.component.css']
+})
+export class AnalysisNamesListComponent implements OnInit {
+
+  @Input() AnalysisType: string;
+  @Input() AnalysisNames: string[];
```
Comment (Sebaxp95) [inline src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.ts]: Variable should start with lower case letter

File: src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+@Component({
+  selector: 'app-analysis-types-list',
+  templateUrl: './analysis-types-list.component.html',
+  styleUrls: ['./analysis-types-list.component.css']
+})
+export class AnalysisTypesListComponent implements OnInit {
+
+  panelOpenState : Boolean[] = [false, false, false];
+  algorithmsUrl = 'http://localhost:2003/algorithms';
```
Comment (Sebaxp95) [inline src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.ts]: Please move urls to apiUrl.ts file

File: src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts (hunk 1 above)
Comment (KapRom) [reply src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts]: It was necessary for the test to see _describe_ , _beforeEach_, _it_. Only in this file, don't know why.

**Comment by gmrukwa** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/spectre-team/spectre) found an issue: Duplicate Code

It's currently on:
[src\app\analysis-types-list\analysis-names-list\analysis-names-list.component.spec.ts:62-74
](https://github.com/spectre-team/spectre/blob/49a004388232dec45e1c87c9000d1cd2234d8df3/src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts#L62-L74)[src\app\analysis-types-list\analysis-types-list\analysis-types-list.component.spec.ts:43-55
](https://github.com/spectre-team/spectre/blob/49a004388232dec45e1c87c9000d1cd2234d8df3/src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.spec.ts#L43-L55)Commit 49a004388232dec45e1c87c9000d1cd2234d8df3

File: src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/*
+* analysis-names-list.component.ts
+* Module for analysis types list.
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.ts]: It's not a module :)

File: src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
+/*
+ * analysis-names-list.component.ts
+ * Module for analysis names list.
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-names-list/analysis-names-list.component.spec.ts]: It's not a module.

File: src/app/analysis-types-list/analysis-types-list.module.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {
+  MatListModule, MatExpansionModule, MatButtonModule,
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-types-list.module.ts]: If you're making it multiline, make it one-per-line.

File: src/app/analysis-types-list/analysis-types-list.service.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Rx';
+import 'rxjs/Rx';
+
+@Injectable()
+export class AnalysisTypesListService {
+
+  constructor(private client: HttpClient) { }
+
+  getAlgorithms(algorithmsUrl: string): Observable<any> {
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-types-list.service.ts]: I think that construction of the URL should be inside this service, not in the component.

File: src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    <i
+      class="fa fa-cogs"
+      aria-hidden="true">
+    </i>
+    <div class="font-cont">
+      {{type}}
+    </div>
+  </mat-panel-title>
+  <mat-panel-description>
+    {{panelOpenState[i] ? 'less...' : 'more...'}}
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.html]: These dots with current width of the sidebar are pretty wide. We can go for simple *less* and *more* without dots.

File: src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+import {MainPageComponent} from "../../main-page/main-page.component";
+import {AnalysisViewComponent} from "../../analysis-view/analysis-view/analysis-view.component";
+import {PageNotFoundComponent} from "../../page-not-found/page-not-found.component";
+import {NewAnalysisViewComponent} from "../../new-analysis-view/new-analysis-view/new-analysis-view.component";
+import {UploadComponent} from "../../upload/upload.component";
+import {FormsModule, ReactiveFormsModule} from "@angular/forms";
+import {GenericFormModule} from "../../generic-form/generic-form.module";
+import {AnalysisTypesListService} from "../analysis-types-list.service";
+import {Observable} from "rxjs/Observable";
+import {Service} from "../../app.service";
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.spec.ts]: Please validate imports.

File: src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+export class AnalysisTypesListComponent implements OnInit {
+
+  panelOpenState : Boolean[] = [false, false, false];
+  algorithmsUrl: string;
+  algorithms: any;
+  types: string[];
+  isTableEmpty = false;
+
+  constructor(private fetchService: AnalysisTypesListService, private urlProvider: Service) {
+    const analysisApiUrlBase = this.urlProvider.getBaseAnalysisApiUrl();
```
Comment (gmrukwa) [inline src/app/analysis-types-list/analysis-types-list/analysis-types-list.component.ts]: Move this to the service. Components should be responsible only for updating view.

File: src/app/analysis-view/analysis-view/analysis-view.component.spec.ts [hunk 12]
```diff
@@ -42,7 +42,7 @@ describe('AnalysisViewComponent', () => {
         RouterModule,
         RouterTestingModule.withRoutes([]),
         MatButtonModule,
-        MatCardModule,
+        MatCardModule
```
Comment (gmrukwa) [inline src/app/analysis-view/analysis-view/analysis-view.component.spec.ts]: Leave this comma untouched! :D

File: src/app/app.component.css [hunk 13]
```diff
@@ -55,14 +57,25 @@ height: 50px;
...
 .fa-cloud-upload
 {
+  margin-left: 8px;
+  margin-right: 5px;
   color: #F69736;
 }
+
+.app-content
+{
+  height: 100vh;
```
Comment (gmrukwa) [inline src/app/app.component.css]: At last! Good point! We'll have to get this a little bit more reasonably in the near future, but for now it's good.

File: src/app/app.component.spec.ts [hunk 14]
```diff
@@ -28,8 +28,24 @@ import { PreparationListComponent } from './preparations/preparation-list/prepar
...
+import {MainPageComponent} from "./main-page/main-page.component";
+import {UploadComponent} from "./upload/upload.component";
+import {PageNotFoundComponent} from "./page-not-found/page-not-found.component";
+import {FormsModule, ReactiveFormsModule} from "@angular/forms";
+import {AnalysisViewComponent} from "./analysis-view/analysis-view/analysis-view.component";
+import {NewAnalysisViewComponent} from "./new-analysis-view/new-analysis-view/new-analysis-view.component";
+import {GenericFormModule} from "./generic-form/generic-form.module";
+import {HttpClientModule} from "@angular/common/http";
+import {AnalysisNamesListComponent} from "./analysis-types-list/analysis-names-list/analysis-names-list.component";
+import {Service} from "./app.service";
```
Comment (gmrukwa) [inline src/app/app.component.spec.ts]: I don't think we need all of these imports.

**Comment by gmrukwa** [comment]
[Create generic analyses list panel](https://trello.com/c/0D5nZdW7/416-create-generic-analyses-list-panel)

**Pull request merged**